pkg/controllers/resources/ingresses/syncer.go

```diff
@@ -60,10 +62,42 @@ func (s *ingressSyncer) Sync(ctx *synccontext.SyncContext, pObj client.Object, v
 
 func SecretNamesFromIngress(ingress *networkingv1.Ingress) []string {
 	secrets := []string{}
+	_, extraSecrets := translateAnnotations(ingress.Annotations, ingress.Namespace)
+	secrets = append(secrets, extraSecrets...)
 	for _, tls := range ingress.Spec.TLS {
 		if tls.SecretName != "" {
 			secrets = append(secrets, ingress.Namespace+"/"+tls.SecretName)
 		}
 	}
 	return translator.UniqueSlice(secrets)
 }
+
+var TranslateAnnotations = map[string]bool{
+	"nginx.ingress.kubernetes.io/auth-secret":      true,
+	"nginx.ingress.kubernetes.io/auth-tls-secret":  true,
+	"nginx.ingress.kubernetes.io/proxy-ssl-secret": true,
+}
+
+func translateAnnotations(annotations map[string]string, currentNamespace string) (map[string]string, []string) {
```

Comment:
nit-pick: `currentNamespace` variable name is slightly confusing, because AFAIK that is how we usually refer to the namespace where we create pods, etc. in the host cluster. Perhaps `ingressNamespace` would be better :)

Reply:
oh true, let me fix that!